test/src/test/java/jenkins/security/AgentSecretActionTest.java

```diff
@@ -0,0 +1,85 @@
+package jenkins.security;
+
+
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import hudson.model.Computer;
+import hudson.slaves.DumbSlave;
+import hudson.slaves.SlaveComputer;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import org.htmlunit.FailingHttpStatusCodeException;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.MockAuthorizationStrategy;
+import org.kohsuke.stapler.StaplerRequest2;
+import org.kohsuke.stapler.StaplerResponse2;
+
+public class AgentSecretActionTest {
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    private DumbSlave agent;
+
+    private StaplerRequest2 req;
+    private StaplerResponse2 rsp;
+    private StringWriter stringWriter;
+    private PrintWriter writer;
+
+    private String agentUrl;
+
+    @Before
+    public void setUp() throws Exception {
+        agent = new DumbSlave("test-agent", "/home/jenkins", new hudson.slaves.JNLPLauncher());
```

Comment:
This creates a slave but will not attach it to Jenkins. So the tests are not be able to work with it.